ml-algorithms/src/main/java/org/opensearch/ml/engine/ModelHelper.java

```diff
@@ -168,30 +171,40 @@ public void downloadAndSplit(String taskId, String modelName, String version, St
         }
     }
 
-    private void verifyModelZipFile(String modelZipFilePath) throws IOException {
-        boolean hasModelFile = false;
+    public void verifyModelZipFile(MLModelFormat modelFormat, String modelZipFilePath) throws IOException {
+        boolean hasPtFile = false;
+        boolean hasOnnxFile = false;
         boolean hasTokenizerFile = false;
         try (ZipFile zipFile = new ZipFile(modelZipFilePath)) {
             Enumeration zipEntries = zipFile.entries();
             while (zipEntries.hasMoreElements()) {
                 String fileName = ((ZipEntry) zipEntries.nextElement()).getName();
-                if (fileName.endsWith(PYTORCH_FILE_EXTENSION) || fileName.endsWith(ONNX_FILE_EXTENSION)) {
-                    if (hasModelFile) {
-                        throw new IllegalArgumentException("Find multiple model files, but expected only one");
-                    }
-                    hasModelFile = true;
-                }
+                hasPtFile = hasModelFile(modelFormat, MLModelFormat.TORCH_SCRIPT, PYTORCH_FILE_EXTENSION, hasPtFile, fileName);
+                hasOnnxFile = hasModelFile(modelFormat, MLModelFormat.ONNX, ONNX_FILE_EXTENSION, hasOnnxFile, fileName);
                 if (fileName.equals(TOKENIZER_FILE_NAME)) {
```

Comment:
We allow multiple tokenizers here?

Reply:
It doesn't matter, we only read the file with name `tokenizer.json`.